server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -208,6 +209,67 @@ public void testPaginationRequiresVerboseListing() throws Exception {
         );
     }
 
+    public void testExcludePatterns() throws Exception {
```

Comment:
Can we add a test validating that using `-xyz` finds such a snapshot (and repo) if no wildcard is specified first, i.e., at that we can still find snapshots/repos starting with `-`?

Reply:
Good point! -> added a couple of tests around this scenario now.